### ISSUE TITLE
Don't recalculate stuff in a code fix

### ISSUE THREAD

**Issue by Vannevelj** [Vannevelj/VSDiagnostics#351]
I know some analyzer+codefix combos re-calculate things that were calculated in the analyzer. Back when those were written, we couldn't pass data between the two. Now we can! Find whatever analyzers that can benefit from this and refactor them.

**Comment by Hosch250** [comment]
Actually, most of mine do.  However, how much can we pass in a string dictionary?

**Comment by Vannevelj** [comment]
Fair enough point. I think it is worth considering passing data through JSON format (this is serialization after all). We'll have to look at some more specific scenarios for that though

**Comment by Hosch250** [comment]
Sure.  I've never liked recalculating all that data.
